src/header.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/sh
+
+set -eu
+TEMPDIR=$(mktemp -d)
+PREFIX=""
+FORCE=0
+VERBOSE=0
+QUIET=0
+UNPACK_SHELL=""
+
+USAGE="
+usage: $0 [options]
+
+Unpacks an environment packed using pixi-pack
+
+-f, --force                 No error if environment already exists
+-h, --help                  Print this help message and exit
+-o, --output-directory <DIR> Where to unpack the environment
+-s, --shell <SHELL>         Sets the shell [options: bash, zsh, xonsh, cmd, powershell, fish, nushell]
+-v, --verbose               Increase logging verbosity
+-q, --quiet                 Decrease logging verbosity
+"
+# Parse command-line options
+while getopts ":hfvo:s:q" opt; do
+  case ${opt} in
+    h )
+      echo "$USAGE"
+      exit 0
+      ;;
+    f )
+      FORCE=1
+      ;;
+    v )
+      VERBOSE=1
+      ;;
+    o )
+      PREFIX="$OPTARG"
+      ;;
+    s )
+      UNPACK_SHELL="$OPTARG"
+      ;;
+    q )
+      QUIET=1
+      ;;
+    \? )
+      echo "Invalid option: -$OPTARG" >&2
+      echo "$USAGE" >&2
+      exit 1
+      ;;
+    : )
+      echo "Option -$OPTARG requires an argument" >&2
+      echo "$USAGE" >&2
+      exit 1
+      ;;
+  esac
+done
+shift $((OPTIND -1))
+
+# Validate shell option if provided
+if [ -n "$UNPACK_SHELL" ]; then
+  case "$UNPACK_SHELL" in
+    bash|zsh|xonsh|cmd|powershell|fish|nushell)
+      ;;
+    *)
+      echo "Invalid shell option: $UNPACK_SHELL" >&2
+      echo "Valid options are: bash, zsh, xonsh, cmd, powershell, fish, nushell" >&2
+      exit 1
+      ;;
+  esac
+fi
+
+if [ "$FORCE" = "0" ] && [ -n "$PREFIX" ] && [ -e "$PREFIX" ]; then
+    echo "ERROR: File or directory already exists: '$PREFIX'" >&2
+    echo "If you want to update an existing environment, use the -f option." >&2
+    exit 1
+fi
+
+if [ "$FORCE" = "1" ] && [ -n "$PREFIX" ] && [ -e "$PREFIX" ]; then
+    rm -rf "$PREFIX"
+fi
+
+archive_begin=$(($(grep -anm 1 "^@@END_HEADER@@" "$0" | sed 's/:.*//') + 1))
+archive_end=$(($(grep -anm 1 "^@@END_ARCHIVE@@" "$0" | sed 's/:.*//') - 1))
```

Comment:
This can be replaced with awk (I already pushed replacement) but awk on linux and macos are still not the same thing. I tested it on both platforms and it seems to be working fine but I guess if we want to make sure they are exactly the same we'll have to use gawk on linux and nawk on macos. I would say this won't turn out very beautiful but is definitely possible.